src/components/hooks/use-handle-lazy-load.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { ReactNode, useEffect, useState } from 'react'
+import { DotLoader } from 'components/elements'
+
+export const useHandleLazyLoad = (lazyTemplates: ReactNode, targetId: string, options: object) => {
+    const [is_visible, setIsVisible] = useState(false)
+    const [is_large_screen, setLargeScreen] = useState(false)
+
+    const handleIntersect = (entries, observer) => {
+        const [entry] = entries
+        if (entry.isIntersecting) {
+            setIsVisible(entry.isIntersecting)
+            observer.disconnect()
+        }
+    }
+    useEffect(() => {
+        createObserver()
+        if (window.innerHeight >= 990) {
+            setLargeScreen(true)
+        }
+    }, [])
```

Comment:
please do the cleanup in the return statement of it 
and also please make sure that it doesn't throw any warning regarding empty dependency array